src/main/java/com/google/devtools/common/options/RegexPatternOption.java

```diff
@@ -29,11 +31,19 @@
 @AutoValue
 public abstract class RegexPatternOption {
   static RegexPatternOption create(Pattern regexPattern) {
-    return new AutoValue_RegexPatternOption(Preconditions.checkNotNull(regexPattern));
+    return new AutoValue_RegexPatternOption(
+        Preconditions.checkNotNull(regexPattern), RegexUtil.asOptimizedMatchingPredicate(regexPattern));
   }
 
+  /** The original regex pattern. */
   public abstract Pattern regexPattern();
```

Comment:
@tjgq This made me realize that the internal string encoding can cause Unicode matchers in regex to not match as expected. To be fully correct, we would need to reencode all paths before matching them and also reencode the pattern. Do you think that's relevant enough to change? (just to be clear, this is preexisting)

Reply:
https://github.com/bazelbuild/bazel/issues/26070